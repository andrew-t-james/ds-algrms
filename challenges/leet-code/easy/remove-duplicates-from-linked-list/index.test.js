const { deleteDuplicates, LinkedList } = require('.')

test('should return unchanged list when no values repeated', () => {
  const list = new LinkedList()
  const nodeList = [1, 2, 3]

  for (let node of nodeList) {
    list.insert(node)
  }

  expect(deleteDuplicates(list.head)).toMatchSnapshot()
})

test('should return updated list when values repeated', () => {
  const list = new LinkedList()
  const nodeList = [1, 2, 2, 3]

  for (let node of nodeList) {
    list.insert(node)
  }

  expect(deleteDuplicates(list.head)).toMatchSnapshot()
})

test('should return handle N repeated values', () => {
  const list = new LinkedList()
  const nodeList = [1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 3]

  for (let node of nodeList) {
    list.insert(node)
  }

  expect(deleteDuplicates(list.head)).toMatchSnapshot()
})

test('should handle case when head is null', () => {
  const list = new LinkedList()

  expect(deleteDuplicates(list.head)).toBe(null)
})
