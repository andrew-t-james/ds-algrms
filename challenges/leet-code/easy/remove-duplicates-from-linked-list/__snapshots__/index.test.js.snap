// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return handle N repeated values 1`] = `
ListNode {
  "next": ListNode {
    "next": ListNode {
      "next": null,
      "val": 3,
    },
    "val": 2,
  },
  "val": 1,
}
`;

exports[`should return unchanged list when no values repeated 1`] = `
ListNode {
  "next": ListNode {
    "next": ListNode {
      "next": null,
      "val": 3,
    },
    "val": 2,
  },
  "val": 1,
}
`;

exports[`should return updated list when values repeated 1`] = `
ListNode {
  "next": ListNode {
    "next": ListNode {
      "next": null,
      "val": 3,
    },
    "val": 2,
  },
  "val": 1,
}
`;
